### ISSUE TITLE
Make segments enumerable

### ISSUE THREAD

**Issue by evanmthw** [ruby-hl7/ruby-hl7#55]
Messages are enumerable on their segments.
```
raw_input = open( "message" ).readlines

msg = HL7::Message.new( raw_input )

#Enumerate segments
msg.each do |seg|
  #One way to enumerate fields
  for i in 0..seg.length
    print seg.send("e"+i.to_s)
  end
  #Proposed enumerable method
  #seg.each do |f|
  #  print f
  #end
end
```
Why are segments not enumerable on their fields?
I am fairly new to HL7 processing so if this doesn't make sense to add then could someone explain why.

**Comment by evanmthw** [comment]
Forgot to mention. I would be glad to make a pull request for this.

**Comment by Stratus3D** [comment]
@evanmthw I think that idea is that segment fields are normally named, so you should be able to just grab the fields you need by name. Enumeration is typically not needed. But I don't disagree, it would be nice to be able to enumerate over fields. 

@ruby-hl7/maintainers I haven't used ruby-hl7 in a long time. Do you guys have any thoughts on this? Was there a reason we didn't implement this?

**Comment by mogox** [comment]
Fixed by #56
